sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -919,23 +916,16 @@ def _migrate_rows(self) -> None:
             return
 
         def map_data_versions(
-            name: str, versions: t.Sequence[SnapshotDataVersion]
+            name: str, versions: t.Tuple[SnapshotDataVersion, ...]
         ) -> t.Tuple[SnapshotDataVersion, ...]:
-            version_ids = ((version.snapshot_id(name), version) for version in versions)
-
-            return tuple(
-                new_snapshots[snapshot_id_mapping[version_id]].data_version
-                if version_id in snapshot_id_mapping
-                else version
-                for version_id, version in version_ids
-            )
+            if versions and versions[-1].snapshot_id(name) in snapshot_id_mapping:
+                new_snapshot = new_snapshots[snapshot_id_mapping[versions[0].snapshot_id(name)]]
```

Comment:
We don't need to map every version. We only need to check whether the latest one was mapped and fetch the version history from it.